pex.go

```diff
@@ -28,16 +29,10 @@ type pexEvent struct {
 	f    pp.PexPeerFlags
 }
 
-// Combines the node addr, as required for pp.PexMsg.
-type pexMsgAdded struct {
-	krpc.NodeAddr
-	pp.PexPeerFlags
-}
-
-// Makes generating a PexMsg more efficient.
+// facilitates efficient de-duplication while generating PEX messages
 type pexMsgFactory struct {
-	added   map[string]pexMsgAdded
-	dropped map[string]krpc.NodeAddr
+	added   map[addrKey]pexEvent
```

Comment:
I don't think I like having the pexEvent as values in `added` and `dropped`. By the time they're stored there, we've already processed the event type, so it's redundant data. I'm left wondering why it's there.

Reply:
It's even worse for dropped events, since the flags don't mean anything either there.

Reply:
I can see your concern: there's an overhead in storing those entries wrapped as an event. However, there's a win, too: the resulting code is cleaner, with much reduced exposure to the `krpc.*` types which reflect how the data is represented on the wire and are clumsy to deal with outside of the serialisation primitives. Consider this an investment into simplicity. It will pay back, eventually.